Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -574,6 +634,61 @@ void configureGetBookmarksDelegate(TestWebExtensionManager *manager)
     [manager loadAndRun];
 }
 
+TEST_F(WKWebExtensionAPIBookmarks, BookmarksAPIRemoveAndRemoveTree)
+{
+    auto script = @[
+        @"let folder1 = await browser.bookmarks.create({ title: 'Folder One' });",
+        @"let bookmarkA = await browser.bookmarks.create({ parentId: folder1.id, title: 'Bookmark A', url: 'http://example.com/a' });",
+        @"let folder2 = await browser.bookmarks.create({ parentId: folder1.id, title: 'Folder Two' });",
+        @"let bookmarkB = await browser.bookmarks.create({ parentId: folder2.id, title: 'Bookmark B', url: 'http://example.com/b' });",
+        @"let topLevelBookmark = await browser.bookmarks.create({ title: 'Top Level Bookmark', url: 'http://example.com/top' });",
+        @"let tree1 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(2, tree1[0].children.length, 'folder1 should have 2 children');",
+        @"browser.test.assertEq('Top Level Bookmark', tree1[1].title, 'Second top-level is topBookmark');",
+        @"browser.test.assertEq('Bookmark A', tree1[0].children[0].title, 'bookmarkA is child of folder1');",
+        @"await browser.test.assertRejects(",
+        @"browser.bookmarks.removeTree(bookmarkA.id),",
+        @"/Can't remove a non-folder item '.*' with bookmarks.removeTree\\(\\). Use bookmarks.remove\\(\\)./,",
+        @"'FAIL-1: removeTree on a bookmark should fail with the correct message');",
+        @"await browser.test.assertRejects(",
+        @"browser.bookmarks.remove(folder2.id),",
+        @"/Bookmark with ID '.*' is a non-empty folder and cannot be removed with bookmarks.remove\\(\\). Use bookmarks.removeTree\\(\\)./,",
+        @"'FAIL-2: remove on a non-empty folder should fail with the correct message');",
+        @"await browser.test.assertRejects(",
+        @"browser.bookmarks.remove('nonexistent-id-123'),",
+        @"/Bookmark with ID 'nonexistent-id-123' not found./,",
+        @"'FAIL-3a: remove on a non-existent ID should fail with the correct message');",
+        @"await browser.test.assertRejects(",
+        @"browser.bookmarks.removeTree('nonexistent-id-456'),",
+        @"/Bookmark with ID 'nonexistent-id-456' not found./,",
+        @"'FAIL-3b: removeTree on a non-existent ID should fail with the correct message');",
+        @"await browser.bookmarks.remove(bookmarkA.id);",
+        @"let tree2 = await browser.bookmarks.getTree();",
+        @"browser.test.assertEq(1, tree2[0].children.length, 'folder1 should now have 1 child after removing bookmarkA');",
+        @"browser.test.assertEq('Folder Two', tree2[0].children[0].title, 'folder2 should be the only child of folder1');",
+        @"await browser.bookmarks.removeTree(folder2.id);",
+        @"let tree3 = await browser.bookmarks.getTree();",
```

Comment:
Thanks for adding all of these tests. I think this is a big enough block of text that it could use some blank lines between test cases for readability (assuming that still compiles).

Reply:
Like, especially around lines 650-652, where it's one line of code that is three lines of one API call.